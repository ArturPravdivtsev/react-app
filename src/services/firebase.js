import firebase from 'firebase'

const firebaseConfig = {
    apiKey: "AIzaSyAv_ENkSTf2XGnK_5Y-D7FUHcHnKTTobnQ",
    authDomain: "react-chat-5e32c.firebaseapp.com",
    databaseURL: "https://react-chat-5e32c-default-rtdb.europe-west1.firebasedatabase.app",
    projectId: "react-chat-5e32c",
    storageBucket: "react-chat-5e32c.appspot.com",
    messagingSenderId: "500405768636",
    appId: "1:500405768636:web:963f6500eabb650afcdbb9",
    measurementId: "G-RHCN7Y0Q3H"
};

firebase.initializeApp(firebaseConfig);