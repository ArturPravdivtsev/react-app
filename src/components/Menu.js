import React from 'react'
import { Link } from 'react-router-dom'
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import firebase from 'firebase'
import { useDispatch } from 'react-redux'
import { changeIsAuthed } from '../store/actions/profile'

export default function Menu() {
    const [open, setOpen] = React.useState(false);
    const handleDrawerOpen = () => {
        setOpen(true);
    };
    const handleDrawerClose = () => {
        setOpen(false);
    };
    const dispatch = useDispatch();
    React.useEffect(() => {
        firebase.auth().onAuthStateChanged((user) => {
            dispatch(changeIsAuthed(Boolean(user)))
        })
    }, []);
    const handleSignOut = (e) => {
        e.preventDefault()
        firebase.auth().signOut()
    };
    return (
        <React.Fragment>
            <AppBar
                position="relative"
            >
                <Toolbar>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        onClick={handleDrawerOpen}
                        edge="start"
                    >
                        <MenuIcon />
                    </IconButton>
                </Toolbar>
            </AppBar>
            <Drawer
                variant="persistent"
                anchor="left"
                open={open}
            >
                <div>
                    <IconButton onClick={handleDrawerClose}>
                        <ChevronRightIcon />
                    </IconButton>
                </div>
                <List component="nav">
                    <ListItem component={Link} to="/" divider button>
                        Home
                    </ListItem>
                    <ListItem component={Link} to="/chats" divider button>
                        Chats
                    </ListItem>
                    <ListItem component={Link} to="/profile" divider button>
                        Profile
                    </ListItem>
                    <ListItem component={Link} to="/news" divider button>
                        News
                    </ListItem>
                    <ListItem component={Link} to="/login" divider button>
                        Login
                    </ListItem>
                    <ListItem component={Link} to="/" divider button onClick={handleSignOut}>
                        Sign out
                    </ListItem>
                </List>
            </Drawer>
        </React.Fragment>
    )
}