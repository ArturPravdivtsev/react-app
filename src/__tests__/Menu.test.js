import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Menu from '../components/Menu';
import firebase from '../services/firebase';

test('Menu matches snapshot', () => {
    const component = render(<Provider store={{getState: jest.fn(), subscribe: jest.fn(), dispatch: jest.fn()}}><BrowserRouter><Menu /></BrowserRouter></Provider>);
    expect(component).toMatchSnapshot();
});
