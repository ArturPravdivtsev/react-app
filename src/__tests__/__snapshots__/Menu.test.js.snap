// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            id="open"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
      <div
        class="MuiDrawer-root MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
          style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
        >
          <div>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <nav
            class="MuiList-root MuiList-padding"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
              href="/"
              role="button"
              tabindex="0"
            >
              Home
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
              href="/chats"
              role="button"
              tabindex="0"
            >
              Chats
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
              href="/profile"
              role="button"
              tabindex="0"
            >
              Profile
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
              href="/news"
              role="button"
              tabindex="0"
            >
              News
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
              href="/login"
              role="button"
              tabindex="0"
            >
              Login
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
              href="/"
              role="button"
              tabindex="0"
            >
              Sign out
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </nav>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          id="open"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
      >
        <div>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <nav
          class="MuiList-root MuiList-padding"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            href="/"
            role="button"
            tabindex="0"
          >
            Home
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            href="/chats"
            role="button"
            tabindex="0"
          >
            Chats
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            href="/profile"
            role="button"
            tabindex="0"
          >
            Profile
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            href="/news"
            role="button"
            tabindex="0"
          >
            News
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            href="/login"
            role="button"
            tabindex="0"
          >
            Login
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            href="/"
            role="button"
            tabindex="0"
          >
            Sign out
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </nav>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Menu matches snapshot 2`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        id="open"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    class="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <nav
        class="MuiList-root MuiList-padding"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          href="/"
          role="button"
          tabindex="0"
        >
          Home
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          href="/chats"
          role="button"
          tabindex="0"
        >
          Chats
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          href="/profile"
          role="button"
          tabindex="0"
        >
          Profile
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          href="/news"
          role="button"
          tabindex="0"
        >
          News
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          href="/login"
          role="button"
          tabindex="0"
        >
          Login
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-divider MuiListItem-button"
          href="/"
          role="button"
          tabindex="0"
        >
          Sign out
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </nav>
    </div>
  </div>
</div>
`;
